File: GO/Licuadora.go
package main
import "fmt"
type Licuadora struct {
	Electrodomestico
	Velocidad int
	Sirve     bool
}
//Se crea el motodo contructor de licaudora
func NewLicuadora(velocidad int, sirve bool) *Licuadora {
	return &Licuadora{
		Velocidad: velocidad,
		Sirve:     sirve,
	}
}
// Implementamos los métodos abstractos
//Metodo para decir que se encendio
func (l *Licuadora) Encender(encendido bool) {
	l.encendido = encendido
	if encendido {
		fmt.Println("Encendida")
	} else {
		fmt.Println("Apagafo")
	}
}
//metodo para decir que esta funcionando
func (l Licuadora) Funcionar() {
	if l.encendido {
		fmt.Println("La licuadora esta licuando")
	} else {
		fmt.Println("La licuadora no esta licuando")
	}
}

File: Java/poocorredor/Licuadora.java
package poocorredor;

public class Licuadora extends Electrodomestico {
     private boolean sirve;
    private String marca;

    public Licuadora(boolean sirve, String marca) {
        this.sirve = sirve;
        this.marca = marca;
    }
    //Metodo para saber si la licuadora sirve o no
    public void servir(boolean sirve){
        if (sirve == false) {
            System.out.println("No sirve");
        }else{
            System.out.println("Si sirve");
        }

    //Metodos heredados de la clase electrodomestico que son abstractos

}
    @Override
    public void encender(boolean encendido) {
            if (encendido==true) {
            System.out.println("Se enciende");
            }else{
            System.out.println("se apaga");
            }
    }
    @Override
    public void funcionar() {
           System.out.println("Funciono");
    }
}
   
    


